### PR TITLE
Prep work for derby tests

### LINKED ISSUE #2430: Tweetest Prep work for derby tests
**👾 Description of the issue**

We run external third-party application tests at this project.  They are contained within the external directory.  We would like to improve how we run these tests.  The first step to improving how we run them is to add some documentation and to pull out the individual test properties that we use so that they are stored in the same directory as the test.

This issue is to add a README.md and a test.properties file to the /external/derby test directory.  As part of this prep work, since we already generate the Dockerfile, please also remove the static https://github.com/AdoptOpenJDK/openjdk-tests/blob/master/external/derby/dockerfile/Dockerfile.

The contents of the README file can be simply some instructions for how to run the test locally.  Use with the instructions from https://github.com/AdoptOpenJDK/openjdk-tests/blob/master/external/README.md#running-external-tests-locally and modify Steps 7 and 9 to reference derby and derby-test specifically.

The contents of the test.properties file will be a key=value format, copying the information from https://github.com/AdoptOpenJDK/openjdk-tests/blob/master/external/common_functions.sh#L194-L206 to populate it.

Adding these files will not change the current way we run these tests.  A separate change will be made to reference the test.properties file, at which point the contents must match those in the common_functions.sh script.  

Please take the time to try and run the test locally (using the instructions you are adding in the README file).  Some of the external tests are failing.  When you run it locally, you may find that to be the case.  The reviewer can help to advise you if there are any further steps you can take to try and debug the pre-existing failure, if it exists.

**📋 Step by Step**

To solve this issue and contribute a fix you should check the following step-by-step list. A more detailed documentation of the workflow can be found [here](https://github.com/firstcontributions/first-contributions/blob/master/README.md).  Note: that you do not need to add yourself to a Contributors.md file as outlined in the detailed instructions.

- [ ] Claim this issue: Comment below.
- [ ]  Fork the repository in github by simply clicking the 'fork' button.
- [ ]  Check out the forked repository
- [ ]  Create a feature branch for the issue. We do not have any naming definition for branches.
- [ ]  Commit your changes.
- [ ]  Start a Pull Request.
- [ ]  Done 👍 Ask in comments for a review :)
- [ ]  If the reviewer find some missing pieces or a problem they will start a discussion with you and describe the next steps how the problem can be solved.
- [ ]  You did it 🎉 We will merge the fix in the master branch.
- [ ]  Thanks, thanks, thanks for being part of this project as an open source contributor ❤️

**🤔❓ Questions**

If you have any questions just ask us directly in this issue by adding a comment. You can join our community chat at [Slack](https://adoptopenjdk.net/slack.html). Next to this you can find a general manual about open source contributions [here](https://github.com/firstcontributions/first-contributions/blob/master/README.md).

### PR DESCRIPTION
**Pull request by khushishikhu**
## Description
Fixes #2430 
According to the issue, the documentation is written including all the required things asked for in the issue description
Please correct me in the PR if written wrong somewhere.
Thank You!!

### CODE REVIEW

**Comment by khushishikhu** [comment]
@smlambert  Please review this PR

File: external/derby/test.properties [hunk 1]
```diff
@@ -0,0 +1,13 @@
+github_url="https://github.com/apache/derby.git"
+        script="derby-test.sh"
```
Comment (sophia-guo) [inline external/derby/test.properties:2]: Indentation please.

File: external/derby/README.md [hunk 2]
```diff
@@ -0,0 +1,14 @@
+## Running External tests locally
```
Comment (sophia-guo) [inline external/derby/README.md:1]: Please add another section before this:

# External derby Tests

Derby tests are part of the external third-party application tests that help verify that the Adoptium binaries are good by running a variety of Java applications inside of Docker containers. AdoptOpenJDK/openjdk-tests/Issue [#172](https://github.com/AdoptOpenJDK/openjdk-tests/issues/172) lists the applications that we have initially targeted to best exercise the Adoptium binaries. For each application, we choose to run a selection of their functional tests. Quarkus Openshift tests are pulled from the [apache/derby](https://github.com/apache/derby.git) repository.

Comment (sophia-guo) [reply external/derby/README.md:1]: Please change the heading to:
Running External derby tests locally

File: external/derby/README.md [hunk 3]
```diff
@@ -0,0 +1,14 @@
...
+
+1. Download/unpack the SDK that you want to test to your test machine
+1. `export TEST_JDK_HOME=</pathToWhereYouInstalledSDK>` 
+1. `git clone https://github.com/AdoptOpenJDK/openjdk-tests.git` 
+1. `cd openjdk-tests`
+1. `./get.sh`
+1. `cd TKG`
+1. export required environment variables, BUILD_LIST and EXTRA_DOCKER_ARGS (example: `export BUILD_LIST=external/jacoco` and `export EXTRA_DOCKER_ARGS="-v $TEST_JDK_HOME:/opt/java/openjdk"`
+1. `make compile`              (This fetches test material and compiles it, based on build.xml files in the test directories)
+1. `make _derby_test`   (When you defined BUILD_LIST to point to a directory in [openjdk-tests/external](https://github.com/AdoptOpenJDK/openjdk-tests/tree/master/external), then this is a testCaseName from the playlist.xml file within the directory you chose)
```
Comment (sophia-guo) [inline external/derby/README.md:14]: Please add the following as the last section:

When [running these from the command-line](https://github.com/AdoptOpenJDK/openjdk-tests/blob/master/doc/userGuide.md#local-testing-via-make-targets-on-the-commandline), these tests are grouped under a make target called 'external', so 'make external' would run the entire set of tests found in the openjdk-tests/external directory. This is unadvisable! Limit what you compile and run, BUILD_LIST=external/`<someSubDirectory>`, and TARGET=`<testCaseNameFromSubdirPlaylist>`
These tests run regularly and results can be found in [TRSS Third Party Application view](https://trss.adoptopenjdk.net/ThirdPartyAppVie).

File: external/derby/README.md (hunk 2 above)
Comment (smlambert) [inline external/derby/README.md:1]: Before this section add:

# External Derby Tests

Derby tests are a part of the external third-party application tests that help verify that the Adoptium binaries are good by running a variety of Java applications inside of Docker containers. AdoptOpenJDK/openjdk-tests/[Issue #172](https://github.com/AdoptOpenJDK/openjdk-tests/issues/172) lists the applications that we have initially targeted to best exercise the Adoptium binaries. For each application, we choose to run a selection of their functional tests. Derby test material is pulled from the [derby](https://github.com/apache/derby.git) repository.


Then also change the heading "Running External tests locally" to be "Running Derby tests locally"

**Review by smlambert — CHANGES_REQUESTED** [review]
Thanks @khushishikhu - I have requested a few changes.  Also as per the issue description, please delete https://github.com/AdoptOpenJDK/openjdk-tests/blob/master/external/derby/dockerfile/Dockerfile which is no longer used or needed, since Dockerfiles are auto-generated now.

**Comment by khushishikhu** [comment]
@smlambert @sophia-guo  Thank you for reviewing the PR. I have done the suggested changes. Please check now

File: external/derby/README.md [hunk 4]
```diff
@@ -0,0 +1,22 @@
+# External (Third Party Container) Tests
```
Comment (smlambert) [inline external/derby/README.md:1]: Please change
External (Third Party Container) Tests
to
External Derby Tests

File: external/derby/README.md [hunk 5]
```diff
@@ -0,0 +1,22 @@
...
+
+Third Party container tests help verify that the AdoptOpenJDK binaries are *good* by running a variety of Java applications inside of Docker containers. AdoptOpenJDK/openjdk-tests/[Issue #172](https://github.com/AdoptOpenJDK/openjdk-tests/issues/172) lists the applications that we have initially targeted to best exercise the AdoptOpenJDK binaries.  For each application, we choose to run a selection of their functional tests.
```
Comment (smlambert) [inline external/derby/README.md:3]: Change 
Third Party container tests help verify
to 
Derby tests are a part of the external third-party application tests that help verify

**Comment by smlambert** [comment]
@khushishikhu - are you able to make updates to this PR based on the last review comments above?

**Comment by smlambert** [comment]
Marking as stale and closing.  This can be reopened and delivered if there are updates.

**Pull request closed without merging**